evm_arithmetization/src/cpu/kernel/constants/journal_entry.rs

```diff
@@ -11,10 +11,11 @@ pub(crate) enum JournalEntry {
     Refund = 8,
     AccountCreated = 9,
     Log = 10,
+    TransientStorageChange = 11,
 }
 
 impl JournalEntry {
-    pub(crate) const COUNT: usize = 11;
+    pub(crate) const COUNT: usize = 12;
 
```

Comment:
Keep private?

Reply:
For context: https://doc.rust-lang.org/core/mem/fn.variant_count.html may one day become stable, and if this is kept private we can switch to using that without breaking anything for anyone. If this is `pub`, we'll have to support it even when it's not needed.

Reply:
This is still crate-private though

Reply:
Let's make it private-private if we can.

Reply:
I'm still not sure to understand, as long as it's not exposed in the API, having `variant_count` stabilized won't be an issue?

This can indeed be made fully private, but I'm not sure to understand the explanation here.

Reply:
For a `pub` item it's a no-brainer right? The smaller the API the better, especially if we're talking about something that is not *actually* needed (like here: the compiler obviously knows how many variants an enum has, so the fact that we have to explicitly set the count is nothing but a Rust-language wart) and that we *know* is probably going to be changed (like here: when `variant_count` stabilizes, this workaround is no longer needed). 
If this const was `pub` then we'd have to keep it until we make a major release, regardless of `variant_count` being stable or not.

When things are `pub (crate)` it's more of a grey zone and up for interpretation. In my head `pub(crate)` means that the item is "free to use" and is part of how the modules in a crate are supposed to interact with each other. To me, it's not really that different to `pub`, it's just "smaller". If/when a module is refactored into its own crate – something that is quite possible – all of its `pub(crate)` items must become `pub` of course. In some sense `pub(crate)` is "on the road to `pub`" if you will?

There are many arguments for why keeping the boundaries between code units small and clear. This principle applies regardless of what the "code unit" refers to (a crate, an app, a REST API, a module, whatever…).

Reply:
As in the other comment, this changes are out of the scope of this PR

Reply:
Yeah agree @4l0n50.

@dvdplm  your explanation makes sense and seems the right way to go, but my question wasn't about best practices but about why you were saying pub(crate) would be an issue if this was to get stabilized. As long as this is not exposed to end users (pub visibility) then there shouldn't be a need for major release even if we were to change the internals for it.